From 62cc82e6486b9b66b340bbf5fb38b0171fb56a7f Mon Sep 17 00:00:00 2001
From: mnlife <mnlife@126.com>
Date: Fri, 11 Aug 2023 22:25:11 +0800
Subject: [PATCH] ASoC: soc-jack: calling snd_soc_jack_report causes a null pointer access
Git-commit: 62cc82e6486b9b66b340bbf5fb38b0171fb56a7f
Patch-mainline: v6.6-rc1
References: jsc#PED-9993 jsc#PED-10202

When snd_soc_card_jack_new is not called or the call fails,
calling this function causes a null pointer access

Signed-off-by: mnlife <mnlife@126.com>
Link: https://lore.kernel.org/r/20230811142511.6570-1-mnlife@126.com
Signed-off-by: Mark Brown <broonie@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 sound/soc/soc-jack.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/soc-jack.c b/sound/soc/soc-jack.c
index f951acb2ce36..b2cc13b9c77b 100644
--- a/sound/soc/soc-jack.c
+++ b/sound/soc/soc-jack.c
@@ -37,7 +37,7 @@ void snd_soc_jack_report(struct snd_soc_jack *jack, int status, int mask)
 	struct snd_soc_jack_pin *pin;
 	unsigned int sync = 0;
 
-	if (!jack)
+	if (!jack || !jack->jack)
 		return;
 	trace_snd_soc_jack_report(jack, mask, status);
 
-- 
2.43.0

